### PR TITLE
Added Quarto Profiles

### PR DESCRIPTION
**Pull request by trgardos**
Added Quarto profiles for the website, Reveal.js slides and the original book (deprecated). Updated README with new and simplified instructions.

Also added a bit of a debug pre-render script which shows the Quarto related environment variables and the best guess at the profile YAML configuration.

### CODE REVIEW

File: ds701_book/_quarto-slides.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    - 17-Regression-I-Linear.qmd
+    - 18-Regression-II-Logistic.qmd
+    - 19-Regression-III-More-Linear.qmd
+    - 20-Recommender-Systems.qmd
+    - 21-Networks-I.qmd
+    - 22-Networks-II-Centrality-Clustering.qmd
+    #- 23-NN-I-Gradient-Descent.qmd
+    - 24-NN-II-Backprop.qmd
+    - 25-NN-III-CNNs.qmd
+    - "!23-NN-I-Gradient-Descent.qmd" # for now ignore this file since cell 10 is stuck
```
Comment (sladenheim) [inline ds701_book/_quarto-slides.yml:32]: If it is complaining about errors in a code block, you can always have the site render and the error displayed in the writing by including in the code block

#| error: true

see https://quarto.org/docs/computations/execution-options.html

File: ds701_book/_quarto-web.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+website:
+  title: "Tools for data science"
+  back-to-top-navigation: true
+  sidebar:
+    style: "docked"
+    search: true
+    contents:
+    - href: index.qmd
+      text: Home
+    - section: "Preliminaires"
```
Comment (sladenheim) [inline ds701_book/_quarto-web.yml:45]: There is this typo which was fixed in the previous PR. I think somehow it must have been overwritten.

**Review by sladenheim — COMMENTED** [review]
This looks good. I made one comment and a small review to fix a typo that overwrote a previous commit.

File: ds701_book/_quarto-slides.yml (hunk 1 above)
Comment (trgardos) [reply ds701_book/_quarto-slides.yml:32]: Thanks. What was happening was the code block was taking a very long time to complete. I'll go back later and check why.

File: ds701_book/_quarto-web.yml (hunk 2 above)
Comment (trgardos) [reply ds701_book/_quarto-web.yml]: Fixed. Good catch.

**Comment by trgardos** [comment]
Thanks @sladenheim . Ok to merge then?

**Review by sladenheim — APPROVED** [review]
Ok to merge!

**Comment by sladenheim** [comment]
I merged this for us.

**Pull request merged**